File: src/algorithms/majority-element.js
/**
 * Given an array nums of size n, returns the majority element.
 * 
 * The majority element is the element that appears more than [n / 2] times.
 * You may assume that the majority element always exists in the array.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number} - Returns the majority element
 */
function majorityElement (nums) {
  let numCount = {};
  for (let i = 0; i < nums.length; i++) {
    numCount[nums[i]] = (numCount[nums[i]] || 0) + 1;
  }
  for (const num in numCount) {
    if (numCount[num] > nums.length / 2) {
      return Number(num);
    }
  }
}

File: src/algorithms/majority-element.py
def majorityElement(nums):
    """Given an array nums of size n, return the majority element. The majority 
    element is the element that appears more than ⌊n / 2⌋ times. You may assume 
    that the majority element always exists in the array.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns the majority element.
    """
    count = {}
    half = len(nums) // 2
    for num in nums:
        if num in count:
            count[num] += 1
        else:
            count[num] = 1
    for x, y in count.items():
        if y > half:
            return x
